Replace `attr` with `getAttribute` and `setAttribute`

p8 opened the pull request:
This pull request includes a

- [ ] Bug fix
- [ ] New feature
- [ ] Translation

The following changes were made

- This replaces jQuery's `attr` function with the vanilla JavaScript `getAttribute` and `setAttribute`.

This is related to: https://github.com/select2/select2/issues/6196

kevin-brown commented:
Can you rebase this with the latest changes on develop (I merged in your PR to fix the CI, thank you!) and back out the `dist` changes for me? The `dist` changes cause annoying merge conflicts across PRs so I prefer to recompile it separately after I merge batches of changes.

p8 commented:
@kevin-brown Thanks! Sure will do.

p8 committed:
Replace `attr` with `getAttribute` and `setAttribute`

This replaces jQuery's `attr` function with the vanilla JavaScript
`getAttribute` and `setAttribute`.

p8 committed:
Replace `attr` with `getAttribute` and `setAttribute`

This replaces jQuery's `attr` function with the vanilla JavaScript
`getAttribute` and `setAttribute`.

p8 committed:
Replace attr with getAttribute and setAttribute

This replaces jQuery's `attr` function with the vanilla JavaScript
`getAttribute` and `setAttribute`.

p8 committed:
Replace attr with getAttribute and setAttribute

This replaces jQuery's `attr` function with the vanilla JavaScript
`getAttribute` and `setAttribute`.

p8 commented:
@kevin-brown I've rebased and removed the dist changes.

jayaddison commented on src/js/select2/results.js:
```diff
@@ -480,9 +480,11 @@ define([
-      self.getHighlightedResults()
-          .removeClass('select2-results__option--highlighted')
-          .attr('aria-selected', 'false');
+      var $highlighted = self.getHighlightedResults();
+      $highlighted.removeClass('select2-results__option--highlighted');
+      if($highlighted[0] != null) {
+        $highlighted[0].setAttribute('aria-selected', 'false');
+      }
```
Can multiple results be highlighted?

kevin-brown replied:
Yes, so I added a test and fixed that issue.

jayaddison replied:
Thanks @kevin-brown

jayaddison commented on src/js/select2/core.js:
```diff
@@ -563,11 +564,11 @@ define([
     this.$element.off('.select2');
-    this.$element.attr('tabindex',
-    Utils.GetData(this.$element[0], 'old-tabindex'));
+    this.$element[0].setAttribute('tabindex',
+      Utils.GetData(this.$element[0], 'old-tabindex'));
 
     this.$element[0].classList.remove('select2-hidden-accessible');
-    this.$element.attr('aria-hidden', 'false');
+    this.$element[0].setAttribute('aria-hidden', 'false');
```
Is the adjustment in whitespace before `Utils.GetData` here intentional?  It is logically contained within a staement that has been modified, so I can understand if the idea is to communicate/include that.

kevin-brown committed:
Replace remaining removeAttr with removeAttribute in multiple.js

kevin-brown committed:
Fix unintentional whitespace change in setAttribute call

kevin-brown committed:
Fix aria-selected update to apply to all highlighted results, not just first

kevin-brown committed:
Add test for aria-selected reset on all highlighted results on mouseenter
